test/test_ops.py

```diff
@@ -665,6 +665,19 @@ def test_inplace_forward_mode_AD(self, device, dtype, op):
 
         self._forward_grad_helper(device, dtype, op, self._get_safe_inplace(op.get_inplace()))
 
+    # Tests that functions that do not support autograd do not fail unexpectedly
+    @_gradcheck_ops(op_db)
+    def test_nondifferentiable(self, device, dtype, op):
+        # Test only when supports_autograd=True and dtype is float64
+        if op.supports_autograd or dtype.is_complex:
+            self.skipTest("Skipped! Autograd is supported or complex dtype.")
+        if "resize" in op.name:
+            self.skipTest("Skipped! Cannot resize variables that require grad.")
+
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        sample = samples[0]
+        result = op(sample.input, *sample.args, **sample.kwargs)
```

Comment:
Do we want to check that the result does not require gradients?

Reply:
If the result has integer dtype, then it would have `requires_grad=False`. But if the function entry has "not_implemented" in derivatives.yml, then the result would have `requires_grad=True` and would raise an error only when backward is called.
The goal of this test is to verify that forward mode doesn't raise errors.

Reply:
Yes, we really should cleanup the concept of `non-differentiable` vs `not implemented`.